python from PyQt5.QtWidgets import QDialog, QMessageBox from PyQt5 import uic import os from .db_manager import DBManager
class RegistrationForm(QDialog): def init(self, parent=None): super().init(parent) uic.loadUi(os.path.join(os.path.dirname(file), '../forms/RegistrationForm.ui'), self) self.db = DBManager() self.buttonNext.clicked.connect(self.next_state) self.buttonPrev.clicked.connect(self.prev_state) self.buttonFinish.clicked.connect(self.finish_registration) self.update_buttons()
def next_state(self): index = self.stackedWidget.currentIndex() if index < self.stackedWidget.count() - 1: self.stackedWidget.setCurrentIndex(index + 1) self.update_buttons()
def prev_state(self): index = self.stackedWidget.currentIndex() if index > 0: self.stackedWidget.setCurrentIndex(index - 1) self.update_buttons()
def update_buttons(self): index = self.stackedWidget.currentIndex() self.buttonPrev.setEnabled(index > 0) self.buttonNext.setEnabled(index < self.stackedWidget.count() - 1) self.buttonFinish.setEnabled(index == self.stackedWidget.count() - 1) if index == self.stackedWidget.count() - 1: self.plainConfirm.setPlainText(self.collect_summary())
def collect_summary(self): return f"""Тип: {self.comboIncidentType.currentText()} Дата: {self.dateTimeIncident.dateTime().toString()} Место: {self.editLocation.text()} Судно: {self.editVesselName.text()} IMO: {self.editIMONumber.text()} Описание: {self.textDescription.toPlainText()} """
def finish_registration(self): type = self.comboIncidentType.currentText() lat = self.editLat.text() lon = self.editLon.text() description = self.textDescription.toPlainText()
# errors = self.validate_data(type, lat, lon, description) if errors: QMessageBox.warning(self, "Ошибка", "\n".join(errors)) return
# self.db.save_incident(type, float(lat), float(lon), description) QMessageBox.information(self, "Успех", "Происшествие зарегистрировано") self.accept()
def validate_data(self, type, lat, lon, description): errors = [] if not type: errors.append("Выберите тип происшествия") try: float(lat) float(lon) except ValueError: errors.append("Неверные координаты") if not description: errors.append("Введите описание") return errors
